### PR TITLE
[#1168] Define the presentation plan as a closed catalogue versioned apart from the application

### LINKED ISSUE #1168: Define the presentation plan: a closed block catalogue with a schema version of its own
## Context

Everything Discover does travels in one contract, so the contract comes first. A presentation plan is what a run
produces: an ordered set of typed blocks, each carrying its own data and its own citations, plus what the run knows
about its own limits.

Two properties are the whole point of defining it separately from the code that produces it.

**It is closed.** Nine block types exist and a plan holds nothing else. That is what lets the client render it with
ordinary typed UI rather than interpreting anything, and it is what keeps a model from inventing a presentation the
client cannot draw.

**It is versioned independently of the application.** A deployment and a client are updated separately — a browser
bundle is served by the deployment, but a desktop head is not — so a client will meet a plan produced by a newer
service. A schema version on the plan, and a version on each block, let that client render what it knows, say what it
cannot, and keep the rest of the run. Without it the only safe response to an unfamiliar block is to discard the whole
answer.

The catalogue and its minimal contracts are given in this issue's parent and are not restated here.

## User stories

**As an agent building a renderer** I want a typed contract with a version on it, **so that** a service ahead of my
client degrades one block rather than the run.

**As somebody reading a result** I want every block to carry its own sources, **so that** checking a fact does not
mean checking the whole answer.

**As a reviewer** I want the catalogue to be closed and enumerated in code, **so that** adding a block type is a
reviewed change rather than a string a model chose.

## Acceptance

- The presentation plan is a typed contract carrying an ordered set of blocks, each of the nine catalogued types with
  the minimal contract its parent gives.
- The plan carries a schema version independent of the application's version, and each block carries its own version.
- The block type set is a closed enumeration in code rather than an open string, and adding a member is a source
  change.
- Citations are a first-class part of the contract: a block references citation identifiers, and a citation resolves
  to a message, a fragment, or an attachment.
- The contract can express that a fact is unsupported, that sources conflict, and that the data behind a block is
  stale.
- A plan holds no markup, no code, and no client-side expression of any kind, and that is enforced by the type rather
  than by review.
- The contract is serialized by the source-generated path both ends already use, and appears in the OpenAPI document
  and the golden contract.

## Out of scope

- Producing a plan, which is the planning child.
- Rendering one.

## Dependencies

First child of #1167; every other child of it depends on this contract.

### PR DESCRIPTION
**Pull request by Krzysztof318**
Closes #1168

## What this adds

The contract a Discover run produces and a client draws, in `backend/src/Application/Discovery/Presentation/`.
It lands in `Application` because that is the only boundary both ends may reference: `AI` will produce a plan
(#1169) and `Host` will serve one (#1170), and neither may depend on the other.

A plan holds an ordered set of blocks drawn from a catalogue of nine types, the citations those blocks rest on,
and what the run knows about its own reach. The nine are `answer`, `evidenceList`, `timeline`, `factTable`,
`people`, `threadState`, `attachmentGallery`, `draft`, and `suggestedAction`, each with the minimal contract the
parent issue gives it.

**Closed by construction rather than by review.** The block hierarchy is closed by a `private protected`
constructor; every free text is a `PresentationText` that refuses a value which *is* a tag — the shape a model
returning XAML, HTML, or SVG produces — refuses control characters, and is bounded; every other member is a
number, a timestamp, an identity, or a value from a closed set. There is no member anywhere whose meaning is
"render this". Deserialization goes through the same constructors, so a plan arriving from a producer this
deployment does not control is held to the same rules as one composed in process.

**Versioned apart from the application.** The plan carries `schemaVersion` and each block the version of its own
type. A block's version is written from the catalogue rather than supplied by whatever composed the plan, and it
is checked again on the way in: a block claiming a revision this build does not implement is refused rather than
read as though it were this one, because silently normalizing it would drop the members that revision added while
the block presented as complete.

**Citations are first class.** They are declared once on the plan and referenced by name from the blocks that
rest on them, so two facts drawn from one message are visibly the same source. A reference the plan does not
declare is refused when it is composed and again when it is read. A citation resolves to an email, a persisted
fragment, or an attachment named by its message and position — each by the local identity the client API already
publishes, not by the remote occurrence, which moves when a folder is renamed.

**Unsupported, conflicting, and stale are expressible.** Every block carries support, its citations, and
freshness, and the combination is enforced: a supported block names at least one citation, a conflicting one at
least two, an unsupported one none.

Where a member carries data the catalogue owns — `PresentationBlockType` (identity plus version),
`FactTableColumn` (identity plus value kind) — the type is a closed enumeration per `$closed-enumeration`.
Everywhere else the set is a plain `enum` with `JsonStringEnumConverter<T>`, because compiler-checked
exhaustiveness is worth more there.

## Deviation from one acceptance criterion, stated

The issue asks for the contract to appear "in the OpenAPI document and the golden contract". The OpenAPI document
is generated from the endpoints the host maps, and this child maps none — the route that streams a run is #1170 —
so there is nothing for it to appear in yet.

The golden half is delivered in full, as a fourth public surface,
`backend/tests/PublicSurfaces.UnitTests/presentation-plan-contract.json`: both version numbers, every block
identity with its version, the fact-table column catalogue with each column's value kind, every bound the contract
refuses to exceed, and the JSON schema each part serializes as, exported from the contract's own source-generated
serializer through
`GetJsonSchemaAsNode`. That is the same information the OpenAPI document would have carried, recorded now rather
than when a route exists, so a version a client keys its renderers by cannot move unnoticed in the meantime. The
surface's own remarks, `docs/features/presentation-plan.md`, `backend/tests/AGENTS.md`, and
`docs/architecture/solution-structure.md` each say plainly that OpenAPI inclusion arrives with the run endpoint.

## Breaking changes

None. Every file is new except the four lines that record the fourth golden surface and the table-of-contents
entry; nothing that existed changed shape, so no configuration key, database column, MCP tool, or deployment
contract moves.

A published surface is nonetheless added: `presentation-plan-contract.json` is now one of the four files a diff in
which obliges a changelog entry in the release that carries it.

## Review rounds

**Round 1** raised two P2 findings, both fixed in `46a8676d`, both answered in their threads and resolved.
`PresentationCitation` was the one type in this family with no constructor body, so a citation could be built — and
read off the wire — with no target and an unspecified label; it now validates both the way `PresentationEvidence` and
`PersonEntry` do, with the identifier left to `PresentationPlan`, which is the only place the uniqueness that makes a
name usable can be checked. `EmailAddressJsonConverter` read an address with no ceiling on its length; it now refuses a
token longer than the 320 octets RFC 5321 leaves room for, read off the token rather than the string it decodes to, so
the value is refused before anything expands it. That bound joins the golden surface, which is why the paragraph above
no longer states a count.

## Verification

- `scripts/verify-fast.sh` — pass, 11933 tests, 0 failures.
- `scripts/verify-full.sh` — pass, exit 0, 255 contract assertions, digest recorded.
- `dotnet msbuild .config/CodeCoverage.proj -t:Collect` — aggregate 95.50% (36176/37879), required 85%.
  Line coverage across the new namespace is 1144/1144.
- `scripts/review-obligations.sh` — every page and test it names is in this change. The three JSON converters it
  reports as unnamed by any test are exercised by name through the serializer they are registered on, which is the
  only path production reaches them by, with the refusal cases asserted in `PresentationCitationTargetTests` and
  `PresentationPlanSerializationTests`. Every number above is from a run on `46a8676d`.
- `$check-docs-licenses` — Docs pass, Changelog n/a, Licenses n/a (no package, tool, service, image, or sample
  added).

The integration suite was not run; nothing here reaches a database, a socket, or the file system.

### CODE REVIEW

File: backend/src/Application/Discovery/Presentation/Citations/PresentationCitation.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/// carry the same label and still resolve to different places.
+/// </para>
+/// </remarks>
+/// <param name="Id">The name blocks refer to this citation by.</param>
+/// <param name="Target">What the citation resolves to.</param>
+/// <param name="Label">What a client prints where the source is named.</param>
+public sealed record PresentationCitation(
+    PresentationCitationId Id,
+    PresentationCitationTarget Target,
+    PresentationText Label);
```
[the remark this answers is not in the document]
Comment (Krzysztof318) [reply backend/src/Application/Discovery/Presentation/Citations/PresentationCitation.cs]: Fixed in 46a8676d. `PresentationCitation` is no longer a bare positional record: its constructor calls `ArgumentNullException.ThrowIfNull(target)` and `PresentationRequirement.Specified(label, nameof(label))`, matching `PresentationEvidence` and `PersonEntry`. Both paths are covered — `PresentationCitationTests` for construction, and `Deserialization_ACitationMissingItsTarget_IsRefused` for the wire path the finding describes.

The identifier stays with `PresentationPlan` rather than moving here. What makes a name usable is that the plan declares it once and that no block names anything else, which is a claim about the set that this constructor cannot make; validating it in both places would leave the plan's own guard unreachable. The constructor's remarks say so.

File: backend/src/Application/Discovery/Presentation/EmailAddressJsonConverter.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options)
+    {
+        if (reader.TokenType != JsonTokenType.String)
+        {
+            throw new JsonException($"An address must be a JSON string, but the token was {reader.TokenType}.");
+        }
+
+        if (!EmailAddress.TryCreate(displayName: null, reader.GetString(), out var address))
```
[the remark this answers is not in the document]
Comment (Krzysztof318) [reply backend/src/Application/Discovery/Presentation/EmailAddressJsonConverter.cs:59]: Fixed in 46a8676d. `EmailAddressJsonConverter` now declares `MaxOctets = 320` — what RFC 5321 leaves room for and nothing more, sixty-four octets of local part, the separator, and two hundred and fifty-five of domain — and refuses a longer token before `EmailAddress.TryCreate` sees it.

The bound is read off the token (`reader.ValueSpan` / `reader.ValueSequence`) rather than off the string it decodes to, so an oversized value is refused before anything expands it rather than after. That leaves the ceiling marginally stricter for a value written with escapes, which an address has no reason to be; the remarks state it. The number joins `presentation-plan-contract.json` beside the other fourteen bounds, and both the refusal and an address using the whole bound are asserted in `PresentationPlanSerializationTests`.

**Comment by Krzysztof318** [comment]
Answering the finding with no line to sit on, from the second pass.

**P2 — the body's bound count.** Correct, and fixed: the description no longer states a number. It now says the golden contract records "every bound the contract refuses to exceed", so the sentence cannot go stale the next time a bound is added — which is the failure mode the finding names rather than the count itself. The verification numbers in the body have been refreshed to the run on `46a8676d` at the same time (11933 tests, aggregate 95.50%, 1144/1144 lines across the new namespace), and a "Review rounds" section now records what the first pass changed.

Both P2 findings from the first pass are fixed in `46a8676d`, answered in their threads, and resolved. No thread is open.

**Comment by Krzysztof318** [comment]
Pushed `ad8f0c78`, which carries one thing the second pass named as a candidate and left to the earlier one.

The address fix in `46a8676d` measured the JSON token before decoding it. That left `PresentationText` and `PresentationCitationId` inconsistent with it: both state an exact bound and both decoded the token into a string first, consulting the bound afterwards — the shape the repository's limit rule exists to prevent, on a contract that by definition reads from a producer this deployment does not control. `PresentationJsonBounds` now refuses a token no value of the stated length could reach, before anything decodes it, and the exact count stays with the type because a ceiling read in octets cannot be applied to characters. A four-thousand-character text made entirely of multi-octet characters still reads, and that boundary case is asserted rather than assumed.

The other candidates the second pass listed are left where that pass left them.

Verification on `ad8f0c78`: `scripts/verify-fast.sh` pass (11936 tests, 0 failures); `scripts/verify-full.sh` pass (exit 0, 255 contract assertions); coverage 95.51% aggregate (36184/37887) with 1160/1160 lines across the new namespace.
